clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsSearchProvider.groovy

```diff
@@ -67,6 +78,33 @@ class CatsSearchProvider implements SearchProvider {
       mappings.putAll(provider.urlMappingTemplates.collectEntries { [(it.key): tmpl.createTemplate(it.value)] })
       return mappings
     }
+
+    scheduledExecutorService.scheduleWithFixedDelay(this, 0, 30, TimeUnit.SECONDS)
+  }
+
+  /**
+   * Periodically refresh cache identifiers that can then be searched over in-memory vs. in-redis.
+   *
+   * This is beneficial for sets (like instances) that may have hundreds of thousands of keys.
+   */
+  @Override
+  void run() {
+    log.info("Refreshing Cached Identifiers (instances)")
+    def identifiers = cacheView.getIdentifiers("instances").findResults { String key ->
+      providers.findResults {
+        def v = it.parseKey(key)
+        if (v) {
+          v["_id"] = key
+        }
+
+        return v?.collectEntries {
+          [it.key, it.value.toLowerCase()]
+        }
+      }[0]
+    }
```

Comment:
This is some pretty complex Map of Maps manipulation. Would be good to have a test.